Realm/RLMRealm.mm

```diff
@@ -766,6 +766,32 @@ + (NSUInteger)schemaVersionAtPath:(NSString *)realmPath encryptionKey:(NSData *)
     return RLMRealmSchemaVersion(realm);
 }
 
++ (BOOL)migrationRequiredAtPath:(NSString *)realmPath encryptionKey:(NSData *)key error:(NSError **)outError {
+    RLMRealm *realm = RLMGetThreadLocalCachedRealmForPath(realmPath);
```

Comment:
No, schemaVersionAtPath actually reads from the realm so it needs one for the current thread.